cmdstanpy/utils/stancsv.py

```diff
@@ -380,6 +381,62 @@ def scan_sampling_iters(
         config_dict['ct_max_treedepth'] = ct_max_treedepth
     return lineno
 
+def scan_timing(fd: TextIO, config_dict: Dict[str, Any], lineno: int) -> int:
+    """
+    Scan timing information from the trailing comment lines in a Stan CSV file.
+
+    #  Elapsed Time: 0.001332 seconds (Warm-up)
+    #                0.000249 seconds (Sampling)
+    #                0.001581 seconds (Total)
+
+
+    It extracts the time values and saves them in the config_dict under the key 'timing'
+    as a dictionary with keys 'warmup', 'sampling', and 'total'.
+    Returns the updated line number after reading the timing info.
+
+    :param fd: Open file descriptor positioned at the timing section.
+    :param config_dict: Dictionary to which the timing info is added.
+    :param lineno: Current line number
+    """
+    timing = {}
+    keys = ['warmup', 'sampling', 'total']
+    while True:
+        pos = fd.tell()
+        line = fd.readline()
+        if not line:
+            break
+        lineno += 1
+        stripped = line.strip()
+        if not stripped.startswith('#'):
+            fd.seek(pos)
+            lineno -= 1
+            break
+        content = stripped.lstrip('#').strip()
+        if not content:
+            continue
+        tokens = content.lower().split()
+        if 'elapsed' in tokens[0]:
+            key = 'warmup'
+            try:
+                t = float(tokens[2])
+            except ValueError:
+                raise ValueError(f"Invalid timing value at line {lineno}: {content}")
+        else:
+            if 'sampling' in tokens[2]:
+                key = 'sampling'
+            elif 'total' in tokens[2]:
+                key = 'total'
+            try:
+                t = float(tokens[0])
+            except ValueError:
+                raise ValueError(f"Invalid timing value at line {lineno}: {content}")
+        timing[key] = t
```

Comment:
I think this could be a bit clearer, something like:

```python
        if 'Warm-up' in content:
            key = 'warmup'
            time_str = tokens[2]
        elif 'Sampling' in content:
            key = 'sampling'
            time_str = tokens[0]
        elif  'Total' in content:
            key = 'total'
            time_str = tokens[0]
            
       
        try:
            timing[key] = float(time_str)
        except ValueError as e:
            raise ValueError(f"Invalid timing value at line {lineno}: {content}") from e
```

This is independent of the question of if we want to actually raise the value error here or
instead make this Optional like the type in `mcmc.py` suggests

Reply:
we should be strict - otherwise, consumers of 'time' will have to deal with missing info.

implemented suggestion, added a few more checks.